app/src/conditional_layer.c

```diff
@@ -101,14 +100,20 @@ static int layer_state_changed_listener(const zmk_event_t *ev) {
             if ((zmk_keymap_layer_state() & mask) == mask) {
                 then_layer_state |= BIT(cfg->then_layer);
             }
+            // Same as above, but for the lock status
+            if ((zmk_keymap_layer_locks() & mask) == mask) {
+                layer_locked_by_conditional |= BIT(cfg->then_layer);
```

Comment:
I am more and more leaning toward the `WRITE_BIT` macro from Zephyr's `zephyr/sys/util_macro.h` for these bit operations, e.g.:

```suggestion
                WRITE_BIT(layer_locked_by_conditional, cfg->then_layer, true);
```